### PR TITLE
Created CreatePost component

### LINKED ISSUE #112: Community Page - Posts - Create Post
## Summary
Create a component that we can show on a specific community's page to create a post.

## Technical Details
Client 
Remember to show when the button is pressed.
Look into this MUI component: https://mui.com/material-ui/react-dialog/#full-screen-dialogs 

## Acceptance Criteria
- When the button is pressed this create post slides up/appears
- When the post button is pressed it should post to the server endpoints
- When the x button or sliding down the component it should hide it and show the X community page

## Figma
image

### PR DESCRIPTION
**Pull request by trevortippery**
## Description
For this PR, I created the CreatePost component that will be used in the Community Page with a click of a button.

## Related Issue
This PR closes #112 and should be used in #110.

## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [x] New feature
- [x] Enhancement
- [ ] Documentation update
- [ ] Other (please specify)
      
The CreatePost uses Material UI's dialog component, slide transition, and floating action buttons (FAB).

References to Material UI:
https://mui.com/material-ui/react-dialog/#full-screen-dialogs
https://mui.com/material-ui/react-floating-action-button/

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functionality 
- [ ] Code Documentation 
- [x] Mobile Responsiveness 
- [x] Code testing 
- [x] Other (Sending data to the backend (refactoring), and swipe down feature to close the dialog)

## Screenshots 
Inital button to open CreatePost component
Screenshot 2024-04-11 at 3 27 47 PM

CreatePost component
Screenshot 2024-04-11 at 3 28 05 PM

## Additional Notes
ChatGPT was used to refactor redundancy of Material UI FABs. I used ChatGPT as a reference and did not copy exactly but rather an idea. All data fields and a choice of a marker needs to be selected before the post button will work. I did not include an error message popup when the text fields are empty, and I did not include the swipe down feature to close the dialog. For now, data points are printed to console. Testing was done on the community page.

Link to ChatGPT conversation: https://chat.openai.com/share/02544161-0bdf-4b7e-8644-18d39aa174ff

### CODE REVIEW

**Review by dafneMena — CHANGES_REQUESTED** [review]
Hi the component looks great! 

I noticed that you added the there dots '...' as a marker but I thought that option was getting removed since we weren't exactly sure what it would point to. I might be wrong though? 

Maybe we can check with the team before merging that way we can delete it if needed!

**Comment by trevortippery** [comment]
> Hi the component looks great!
> 
> I noticed that you added the there dots '...' as a marker but I thought that option was getting removed since we weren't exactly sure what it would point to. I might be wrong though?
> 
> Maybe we can check with the team before merging that way we can delete it if needed!

Yeah should be an easy fix. I completely forgot. Thank you for noticing.

**Comment by KingJordan152** [comment]
> Hi the component looks great!
> 
> I noticed that you added the there dots '...' as a marker but I thought that option was getting removed since we weren't exactly sure what it would point to. I might be wrong though?
> 
> Maybe we can check with the team before merging that way we can delete it if needed!

I think we can go ahead and delete that. The options provided are good enough for now.

File: client/src/components/CreatePost/CreatePost.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            console.log('Description:', description);
+            console.log('Date:', date);
+            console.log('Selected Marker:', marker);
+            console.log('Location:', location);
+            handleClose();
+        }
+    }
+
+    return (
+        <React.Fragment>
```
Comment (KingJordan152) [inline client/src/components/CreatePost/CreatePost.tsx:76]: `React.Framgment` is an old React pattern. You can just use empty tags `<></>`.

**Comment by trevortippery** [comment]
Etc marker has been removed from the code.

**Review by KingJordan152 — CHANGES_REQUESTED** [review]
Overall, this component looks super nice, Trevor. Great work on making it look very similar to the Figma!

I have a few requests regarding initializing fields with data automatically and code convention. Also, please fix the conflict you have in `client/src/components/index.ts`.

Other than that, great work!

**Review by Aligary — APPROVED** [review]
Great work, Trevor! Well update sending the post to the backend when it's ready.

**Review by dafneMena — APPROVED** [review]
Looks good, thanks!

**Review by Linkaloo — APPROVED** [review]

File: client/src/components/CreatePost/CreatePost.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    const [date, setDate] = useState('');
+    const [description, setDescription] = useState('');
+    const [location, setLocation] = useState('');
+
+    // Function to handle opening CreatePost component
+    const handleClickOpen = () => {
+        setOpen(true);
+        setPostName('');
+        setMarker(null);
+        setDate('');
```
[the remark this answers is not in the document]
Comment (trevortippery) [reply client/src/components/CreatePost/CreatePost.tsx:46]: Yeah absolutely.

**Comment by trevortippery** [comment]
The changes should be implemented now. I used geolocation and prompt for the user permission to get their current location. The location should include their latitude and longitude.

**Comment by KingJordan152** [comment]
@trevortippery I just learned you can implement the swiping feature from iOS using this MUI component: 
https://mui.com/material-ui/react-drawer/#swipeable

Maybe you should look into implementing this before merging this component in?

File: client/src/components/CreatePost/CreatePost.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            console.log('Description:', description);
+            console.log('Date:', date);
+            console.log('Selected Marker:', marker);
+            console.log('Location:', location);
+            handleClose();
+        }
+    }
+
+    return (
+        <>
```
Comment (KingJordan152) [inline client/src/components/CreatePost/CreatePost.tsx:100]: I think you could replace this fragment with this component from MUI: https://mui.com/material-ui/react-drawer/#swipeable

**Review by KingJordan152 — CHANGES_REQUESTED** [review]
Thank you for implementing my changes. This looks great, thank you for your hard work.

I left one more request for you regarding the swiping-to-dismiss feature. I just found an MUI component that does that for you. If you could implement that in this component that would be perfect!

**Comment by trevortippery** [comment]
Swipe feature for mobile view should be implemented. @KingJordan152

**Review by KingJordan152 — APPROVED** [review]
This looks amazing. Thank you for implementing the swipe feature. Approved!

**Pull request merged**